src/Confluent.SchemaRegistry.Serdes.Protobuf/ProtobufSerializerConfig.cs

```diff
@@ -206,29 +206,30 @@ public static class PropertyNames
 
         /// <summary>
         ///     Reference subject name strategy.
-        ///     
-        ///     default: ReferenceSubjectNameStrategy.ReferenceName
+        ///     The provided type must be an implementation of <see cref="IReferenceSubjectNameStrategy" />
+        ///     default: <see cref="DefaultReferenceSubjectNameStrategy" />
         /// </summary>
-        public ReferenceSubjectNameStrategy? ReferenceSubjectNameStrategy
+        public Type ReferenceSubjectNameStrategy
```

Comment:
Thanks for the PR @aministrator2k 

A problem with this approach is it breaks backward compatibility, so we can't do it.

I'm also not into the idea of using `Type` and `Activator.CreateInstance` here. You've reminded me why I didn't implement this in the initial scope - the analogous thing to do with other parts of the library is to make a `ProtobufSerializerBuilder` class (in this case we would keep the ability to construct `ProtobufSerializer` without the builder as well) and have a set method for this, but that's such a lot of boilerplate... the other thing we could do is just add two additional optional arguments to the constructor for the `SubjectNameStrategy` and `ReferenceSubjectNameStrategy`. That might be more pragmatic. Don't have a strong opinion ATM.

Finally, you've switched from using a delegate to `IReferenceSubjectNameStrategy`. Is there a good reason for that? My preference is for a delegate unless there is good reason it won't be able to satisfy requirements (unsure).

Reply:
Thanks for the feedback @mhowlett.

We share the concern for using `Activator.CreateInstance`. The reasoning behind it for me was to stay as close as possible to Java client design. Same goes for the use of interface.

I like your idea of adding a `ReferenceSubjectNameStrategy` and considered it initially. I gave up that idea because I wasn't sure about the desired direction of the library.

I can work on an alternate pull request by implementing a new `ReferenceSubjectNameStrategy`. The most important thing for us is the support for custom reference names and I see both solutions viable. How does that sound?

Reply:
> The reasoning behind it for me was to stay as close as possible to Java client design.

better off in this case to stay idiomatic to other parts of the .NET client i think.

i'm hopeful i'll have time to slip in doing this myself, but can't guarantee it. i'm erring on the side of making builder classes ATM, but won't have a clear view on it until i try to do it and see how it feels.

I'm interested in your feedback on:

> Finally, you've switched from using a delegate to IReferenceSubjectNameStrategy. Is there a good reason for that?

Reply:
For the use of interface I considered the case of stateful strategies. For example for the case of the strategy we are interested in that have to query schema registry first to grab the full schema and extract registered reference subject names, it is a good idea to cache said schemas in the object.

Reply:
> better off in this case to stay idiomatic to other parts of the .NET client i think.

I agree however the java style that supports injection of custom strategies gives users of the library more freedom and control.

Reply:
> A problem with this approach is it breaks backward compatibility, so we can't do it.

I was wondering if there is a chance for this PR to go through in its current form. To me the risk for breaking backward compatibility is low because the enumeration contained only one value which makes it improbable for someone to try to set it explicitly. And that is the only case this would break old code since the default behavior is the same.